crates/core/src/factories/mem_transport.rs

```diff
@@ -154,23 +166,53 @@ impl TxImp for MemTransport {
     }
 }
 
-type Res = tokio::sync::oneshot::Sender<K2Result<()>>;
+/// A oneshot sender to report back a K2Result.
+type ResultSender = tokio::sync::oneshot::Sender<K2Result<()>>;
+
+/// The sending end of the channel to send commands to a MemTransport's
+/// command runner task.
 type CmdSend = tokio::sync::mpsc::UnboundedSender<Cmd>;
+
+/// The receiving end of the channel to send commands to a MemTransport's
+/// command runner task.
 type CmdRecv = tokio::sync::mpsc::UnboundedReceiver<Cmd>;
-type DataSend = tokio::sync::mpsc::UnboundedSender<(bytes::Bytes, Res)>;
-type DataRecv = tokio::sync::mpsc::UnboundedReceiver<(bytes::Bytes, Res)>;
-type ConSend = tokio::sync::mpsc::UnboundedSender<(Url, DataSend, DataRecv)>;
-type ConRecv = tokio::sync::mpsc::UnboundedReceiver<(Url, DataSend, DataRecv)>;
 
-struct DropSend {
-    send: DataSend,
+/// The sending end of a "data" channel between two MemTransport instances.
+///
+/// Used to send the bytes of a message alongside with a ResultSender to
+/// report back the result of handling the message.
+type DataSend =
+    tokio::sync::mpsc::UnboundedSender<(bytes::Bytes, ResultSender)>;
+
+/// The receiving end of a "data" channel between two MemTransport instances.
+/// Used to send the bytes of a message alongside with a ResultSender to
+/// report back the result of handling the message.
```

Comment:
```suggestion
/// The receiving end of a "data" channel between two MemTransport instances.
///
/// Used to send the bytes of a message alongside with a ResultSender to
/// report back the result of handling the message.
```

I only commented on one of these, but there was another case of the same thing